### ISSUE TITLE
How can solve the problem of action == Nan in PPO？

### ISSUE THREAD

**Issue by niu0717** [MorvanZhou/Reinforcement-learning-with-tensorflow#137]

**Comment by jandevries123** [comment]
if action == Nan:
    action = "random action"

**Comment by ChuaCheowHuan** [comment]
This is most likely caused by exploding gradients in the Actor's network . Try initializing the weights in the actor's network with a non random initializer. 

```
a_w = tf.glorot_uniform_initializer()
mu = tf.layers.dense(self.s, A_DIM, tf.nn.tanh, kernel_initializer = a_w, name='mu', trainable=True)
sigma = tf.layers.dense(self.s, A_DIM, tf.nn.softplus, kernel_initializer = a_w, name='sigma', trainable=True) + 1e-4
```
